prov/efa/src/rdm/efa_rdm_pke_utils.h

```diff
@@ -55,11 +61,21 @@ size_t rxr_pkt_hdr_seg_offset(char *pkt)
 		[RXR_RUNTREAD_TAGRTM_PKT] = offsetof(struct rxr_runtread_rtm_base_hdr, seg_offset),
 	};
 
-	int pkt_type = rxr_get_base_hdr(pkt)->type;
-	int hdr_offset = offset_of_seg_offset_in_header[pkt_type];
+	pkt_type = rxr_get_base_hdr(pke->wiredata)->type;
+	if (efa_rdm_pkt_type_is_eager(pkt_type) ||
```

Comment:
Trying to think of a better way to check this to make it cleaner... Does it mean tthe pkt contains data and it's req? Does `...contain_data(pkt_type) && ...is_req(pkt_type)` work?

Reply:
It is a packet that has data but do not have `seg_offset`. which is `contains_data() && (!is_mulreq) && (!is_data))` I think it is not better than what we have now

Reply:
then we should add a comment or make this check an inline function to say it explicitly

Reply:
good idea. let me do that